src/util/JoinAlgorithms/JoinAlgorithms.h

```diff
@@ -1268,16 +1269,23 @@ struct BlockZipperJoinImpl {
   template <bool DoOptionalJoin>
   void runJoin() {
     fetchAndProcessUndefinedBlocks();
+    if (potentiallyHasUndef && !hasUndef(leftSide_) && !hasUndef(rightSide_)) {
+      // Run the join without UNDEF values if there are none.
+      BlockZipperJoinImpl<LeftSide, RightSide, LessThan, CompatibleRowAction,
+                          AlwaysFalse>{leftSide_, rightSide_, lessThan_,
+                                       compatibleRowAction_, AlwaysFalse{}}
```

Comment:
Can /should you `std::move(...)`here?
I think we have the invariant anyway, that this method can be called only once, so we don't require copyability.